server/src/main/java/keywhiz/service/daos/SecretDAO.java

```diff
@@ -449,6 +489,29 @@ public void deleteSecretsByName(String name) {
         .deleteSecretSeriesByName(name);
   }
 
+  /**
+   * Renames the secret, specified by the secret id, to the name provided
+   * Assumes that there are no secrets with the name provided - i.e. that getSecretsByName
```

Comment:
nit: I'm not sure how to interpret this comment--it actually wouldn't be safe to assume there are no secrets with the name provided, since there could be a race between something else creating the new secret and our call to rename even if we'd called `getSecretsByName` just before calling `renameSecretById`. However, in practice the `UNIQUE` index on secret names means that if there is such a race, one of the two will fail. 

It is good to document that this will throw an exception if the name exists, though, and to suggest calling `getSecretsByName` or something similar to reduce the likelihood of the exception.